great_expectations/datasource/fluent/sqlite_datasource.py

```diff
@@ -209,6 +214,25 @@ class SqliteDatasource(SQLDatasource):
     _TableAsset: Type[SqlTableAsset] = pydantic.PrivateAttr(SqliteTableAsset)
     _QueryAsset: Type[SqlQueryAsset] = pydantic.PrivateAttr(SqliteQueryAsset)
 
+    @override
+    def _create_engine(self) -> sa.engine.Engine:
+        """
+        Create the engine from the connection string, applying config substitutions.
+        Also set the poolclass to StaticPool to avoid issues with multithreading.
+        """
+        model_dict = self.dict(
+            exclude=self._get_exec_engine_excludes(),
+            config_provider=self._config_provider,
+        )
+        _check_config_substitutions_needed(
+            self, model_dict, raise_warning_if_provider_not_present=True
+        )
+        # the connection_string has had config substitutions applied
+        connection_string = model_dict.pop("connection_string")
+        kwargs = model_dict.pop("kwargs", {})
+        # if needed a user could set a different `_poolclass` on the datasource.
+        return sa.create_engine(connection_string, poolclass=self._poolclass, **kwargs)
```

Comment:
https://github.com/great-expectations/great_expectations/blob/92e6fd2128132e1bb27b114dc0c6231c518ddd6b/great_expectations/execution_engine/sqlalchemy_execution_engine.py#L281-L282

Reply:
TIL that this was an option.